pkg/cmd/server/start/start_master.go

```diff
@@ -529,6 +530,10 @@ func startControllers(oc *origin.MasterConfig, kc *kubernetes.MasterConfig) erro
 		return nil
 	}
 
+	if sdnplugin.IsOpenShiftNetworkPlugin(oc.Options.NetworkConfig.NetworkPluginName) && oc.Options.ControllerLeaseTTL <= 0 {
```

Comment:
I don't think this is true.  The lock is only acquired just below, in ControllerPlugStart(), which comes from newControllerPlug(), which end up being "leased.Run()", which is:

```
// Run tries to acquire and hold a lease, invoking Start()
// when the lease is held and invoking Stop() when the lease
// is lost. If the lease was lost gracefully, nil is returned.
// If the lease was lost due to an error, the error is returned.
func (l *Leased) Run() error {
	ch := make(chan error, 1)
	go l.leaser.AcquireAndHold(ch)
	var err error
	defer l.Stop(err)
	for {
		var ok bool
		err, ok = <-ch
		if !ok {
			return nil
		}
		if err != nil {
			for range ch {
				// read the rest of the channel
			}
			return err
		}
		l.Start()
	}
}
```

That's why I put the check before the go func()...